python/nammu/utils/__init__.py

```diff
@@ -136,40 +136,77 @@ def get_yaml_config(yaml_filename):
     return yaml.load(open(path_to_config, 'r'))
 
 
-def update_yaml_config(path_to_jar, yaml_path, path_to_config):
+def compare_version_tuples(version1, version2):
+    """Examine two version tuples. Return 0 if unequal, 1 if equal.
+    http://stackoverflow.com/questions/1714027/version-number-comparison
+    """
+    def normalize(v):
+        return [int(x) for x in re.sub(r'(\.0+)*$', '', v).split(".")]
+    return cmp(normalize(version1), normalize(version2))
+
+
+def update_yaml_config(path_to_jar, yaml_path, path_to_config, verbose=False,
+                       test_mode=False):
     '''
-    Load local config and jar config. Compare versions, if they differ,
-    update local version with newer one.
+    Load local config and jar config. Compare versions. If they differ,
+    it will scan the key-value pairs, and add new key-values not present
+    in the local config. If the same keys are present in the local and jar,
+    the local values will be maintained.
     '''
-    # Load JAR config, or development version if running from console and not
-    # from JAR
     try:
         jar_contents = zipfile.ZipFile(path_to_jar, 'r')
     except zipfile.BadZipfile:
         jar_config = yaml.load(open(path_to_jar, 'r'))
     else:
         jar_config = yaml.load(jar_contents.open(yaml_path))
 
-    # Load local config
     local_config = yaml.load(open(path_to_config, 'r'))
-    # Load version numbers
-    jar_version = jar_config['version']
-    local_version = local_config['version']
-
-    if 'version' in local_config and local_version == jar_version:
-        # Nothing to do, local config is up to date
-        return
+    jar_version = str(jar_config['version'])
+    local_version = str(local_config['version'])
+    diffrent_versions = compare_version_tuples(jar_version, local_version)
+    if diffrent_versions:
+        d = {}
+        if verbose:
+            print("Comparing install and local settings files...")
```

Comment:
We could try and put those prints in the logger, although I'm not entirely sure it's loaded by the time this gets run. We'll have to check.